### DISCUSSION TITLE
Undesired addition of Steiner points

### DISCUSSION THREAD

**Discussion by jeffreystudent** [inducer/meshpy#88] [Q&A]
Hi,

I am working on the re-meshing of a tetrahedron by adding 4 points along edges of the original element. However, an extra point is added every time, and I am pretty sure it is not necessary. Am I missing an option that I am not using?

I tried using the -S switch and setting opt.steinerleft to zero, but that way two points are added. I do know that the shape of the elements will be very narrow but that is okay to me. 

Thanks in advance 

```
nodes  = [[0.76578835, 0.        , 0.40218225],
        [0.87636194, 0.        , 0.57554289],
        [0.67474206, 0.24781317, 0.32118552],
        [0.74987874, 0.25      , 0.71864268]]

addednodes = [[0.67869686, 0.24295228, 0.32617477],
            [0.67755877, 0.24014655, 0.32369132],
            [0.74997873, 0.24842882, 0.71665382],
            [0.75075147, 0.24827502, 0.7176553 ]]

allnodes = nodes + addednodes

allfaces = [[5, 0, 4], [4, 2, 5], [1, 4, 0], [1, 0, 6], [6, 3, 7], [6, 7, 1], [2, 5, 6], [6, 3, 2], [6, 5, 0], [2, 4, 3], [7, 3, 4], [7, 4, 1], [6, 7, 5], [4, 6, 5]]

# Initialize tetgen package
mesh_info = tet.MeshInfo()

# Set optitions for tetgen
# With -S switch we can set steinerleft, but not with -Y switch, which keeps faces intact
opt = tet.Options(switches="pqY")
# opt.steinerleft = 0
# Set nodes and faces
mesh_info.set_points(allnodes)
mesh_info.set_facets(allfaces)

# Create new elements
mesh = tet.build(mesh_info, options=opt)

print( '{} extra nodes added'.format(len(mesh.points) - len(allnodes)))


```

**Comment by inducer** [comment]
As far as I remember, TetGen developers does not have this option, but I think this would be better to ask of the TetGen developers. If the underlyng code has the capability, we can certainly make it available in meshpy.
